asset/encoding.go

```diff
@@ -64,16 +64,16 @@ func VarBytesDecoder(r io.Reader, val any, buf *[8]byte, _ uint64) error {
 
 		// We'll limit all decoded byte slices to prevent memory blow
 		// ups or panics.
-		if bytesLen > (2<<24)-1 {
+		if bytesLen > maxLen {
 			return fmt.Errorf("%w: %v", ErrByteSliceTooLarge,
 				bytesLen)
 		}
 
-		var bytes []byte
-		if err := tlv.DVarBytes(r, &bytes, buf, bytesLen); err != nil {
+		var decoded []byte
+		if err := tlv.DVarBytes(r, &decoded, buf, bytesLen); err != nil {
```

Comment:
Yeah, so `maxLen` is basically the hard coded (most of the times, sometimes implied by the parent record's max size) upper limit we allow for any allocation. Because `bytesLen` is part of the (potentially malicious) payload, we need to limit its maximum. But yes, if the `bytesLen` is below the `maxLen`, we allow allocating as many bytes as requested. If there aren't that many to read, we'll just end up in a `io.EOF` error which is okay.